pyi.py

```diff
@@ -852,39 +851,12 @@ def _check_for_typevarlike_assignments(
             else:
                 self.error(node, Y001.format(cls_name))
 
-    def _Y015_error(self, node: ast.Assign | ast.AnnAssign) -> None:
-        old_syntax = unparse(node)
-        copy_of_node = deepcopy(node)
-        copy_of_node.value = ast.Constant(value=...)
-        new_syntax = unparse(copy_of_node)
-        error_message = Y015.format(old_syntax=old_syntax, new_syntax=new_syntax)
-        self.error(node, error_message)
-
-    @staticmethod
-    def _Y015_violation_detected(node: ast.Assign | ast.AnnAssign) -> bool:
-        assignment = node.value
-
-        if isinstance(node, ast.AnnAssign):
-            if assignment and not isinstance(assignment, ast.Ellipsis):
-                return True
-            return False
-
-        if isinstance(assignment, (ast.Num, ast.Str, ast.Bytes)):
-            return True
-        if (
-            isinstance(assignment, ast.UnaryOp)
-            and isinstance(assignment.op, ast.USub)
-            and isinstance(assignment.operand, ast.Num)
-        ):
-            return True
-        if (
-            isinstance(assignment, (ast.Constant, ast.NameConstant))
-            and not isinstance(assignment, ast.Ellipsis)
-            and assignment.value is not None
-        ):
-            return True
-
-        return False
+    def _is_valid_assignment_value(self, node: ast.expr) -> bool:
+        return (
+            isinstance(node, (ast.Call, ast.Name, ast.Attribute, ast.Subscript))
+            or (isinstance(node, ast.BinOp) and isinstance(node.op, ast.BitOr))
```

Comment:
For extra paranoia could check that the operands to `|` are themselves valid, but fine to omit that.

Reply:
Oh good idea, I'll do that